controllers/azurejson_machine_controller.go

```diff
@@ -99,15 +99,17 @@ func (f filterUnclonedMachinesPredicate) Generic(e event.GenericEvent) bool {
 func (r *AzureJSONMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
 	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultedLoopTimeout(r.ReconcileTimeout))
 	defer cancel()
-	log := r.Log.WithValues("namespace", req.Namespace, "azureMachine", req.Name)
 
-	ctx, span := tele.Tracer().Start(ctx, "controllers.AzureJSONMachineReconciler.Reconcile",
+	ctx, log, done := tele.StartSpanWithLogger(
+		ctx,
+		"controllers.AzureJSONMachineReconciler.Reconcile",
 		trace.WithAttributes(
```

Comment:
What do folks think about leaking upward the `trace.WithAttributes` at this level? If we don't know about the span, we probably don't want to know about the OT trace attributes.

Reply:
what if it was abstracted to API-native attributes and translated below. the API could be something like:

```go
tele.StartSpanWithLogger(ctx, "mything", tele.KVP("namespace", req.Namespace), tele.KVP(...))
```

Reply:
@devigned see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1575/commits/f3aeb5a50f9a8551aac7f13b906f9d13764f8ae3 for generic attributes and LMK what you think.